packages/wallets/particle/src/adapter.ts

```diff
@@ -37,12 +45,46 @@ export class ParticleAdapter extends BaseMessageSignerWalletAdapter {
     private _readyState: WalletReadyState =
         typeof window === 'undefined' ? WalletReadyState.Unsupported : WalletReadyState.Loadable;
 
+    private _particleNetwork: ParticleNetwork | null = null;
+
     constructor(config: ParticleAdapterConfig = {}) {
         super();
         this._connecting = false;
         this._publicKey = null;
         this._wallet = null;
-        this._config = config;
+
+        const defaultNestedConfig: NestedConfig = {
+            projectId: '',
+            clientKey: '',
+            appId: '',
+        };
+
+        const nestedConfig: NestedConfig = {
+            ...defaultNestedConfig,
+            ...config.config,
+        };
+
+        const chainId = nestedConfig.chainId !== undefined ? nestedConfig.chainId : 101;
+        const chainName = nestedConfig.chainName !== undefined ? nestedConfig.chainName : 'solana';
+
+        this._config = {
+            ...config,
+            config: {
+                ...nestedConfig,
+                chainId,
+                chainName,
+                projectId: nestedConfig.projectId,
+                clientKey: nestedConfig.clientKey,
+                appId: nestedConfig.appId,
+            },
+        };
+    }
+
+    public get particle(): ParticleNetwork {
+        if (!this._particleNetwork) {
+            throw new Error('ParticleNetwork is not initialized.');
+        }
+        return this._particleNetwork;
```

Comment:
Can you refactor this so that the getter doesn't throw and just returns the value, which may be `null`?